src/lib/decorators.ts

```diff
@@ -225,6 +225,54 @@ export function query(selector: string) {
   };
 }
 
+/**
+ * A property decorator that converts a class property into a getter that
```

Comment:
Why is this preferable over:

```ts
class {
  @query('#foo') foo;

  x() {
    await this.updateComplete;
    this.foo;
  }
}
```

ie, when would we tell someone to use one pattern over the other?

Reply:
Inside the element, this is arguably just slightly sugared. For a property that needs to be accessed external to the element, using `@asyncQuery` is much cleaner since the user doesn't have to know about `updateComplete`.

Reply:
Do we have use cases where an element wants to expose part of it's internals like this?

Reply:
This is useful when a property that is a query can change with element state. It's preferable when the property is public so that the user does not have to know about the `updateComplete` promise.